### ISSUE TITLE
BehaviorSubject: Optionally omit initial event after `listen()`

### ISSUE THREAD

**Issue by Endebert** [ReactiveX/rxdart#343]
I have a feature request regarding the `BehaviorSubject`. The concept of tracking the latest value is very beneficial for me, however, in combination with e.g. a `StreamBuilder`, I feel the initial event for new listeners of a `BehaviorSubject` can be an unnecessary performance hog.

Take this code for example:
```dart
BehaviorSubject<String> name = BehaviorSubject.seeded('Alice');

class FooWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return StreamBuilder<String>(
      stream: name,
      builder: (context, snapshot) {
        return Text(snapshot.data ?? 'UNKNOWN');
      },
    );
  }
}
```
In this implementation, `StreamBuilder` will always run `builder` twice: 
1. with `snapshot.data == null`, 
2. with `snapshot.data == 'Alice'`.

We could add `initialData: name.value` to `StreamBuilder`, but its `builder` would still be called twice with `snapshot.data == 'Alice'`.

I therefore suggest an optional boolean flag for `BehaviorSubject` factories, that would turn off the initial `listen()` event (which causes the second `builder` execution).

Example: 
```dart
BehaviorSubject<String> name = BehaviorSubject.seeded('Alice', skipInitialEvent: true);

class FooWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return StreamBuilder<String>(
      stream: name,
      initialData: name.value,
      builder: (context, snapshot) {
        return Text(snapshot.data ?? 'UNKNOWN');
      },
    );
  }
}
```
Here, the `StreamBuilder` would call `builder` only once with `snapshot.data == 'Alice'`.

---
What do you think? This might lead to unintended behavior, if people are not aware that `skipInitialEvent` has been set.

The alternative would be to introduce a new class that is very similar to `BehaviorSubject`, but omits the initial event for new listeners.

**Comment by brianegan** [comment]
I agree this is an important use-case, but I would generally favor using `skip(1)` or creating a new class instead of adding an option to `BehaviorSubject`. Why? The replay functionality is part of the spec and I'd imagine it will be confusing: http://reactivex.io/rxjs/manual/overview.html#behaviorsubject.

**Comment by Endebert** [comment]
> I would generally favor using skip(1)...

I did not know this was possible. Are there any pitfalls besides having to add skip(1) every time it's used in conjunction with a StreamBuilder?

> ...or creating a new class instead of adding an option to BehaviorSubject

Any suggestions regarding new class name?

**Comment by rich-j** [comment]
The behavior described above is due to the Flutter [SteamBuilder](https://api.flutter.dev/flutter/widgets/StreamBuilder-class.html) widget. The help section for [initialData](https://api.flutter.dev/flutter/widgets/StreamBuilder/initialData.html) explicitly states:

> ... the first frame will be built with the value null, regardless of whether a value is available on the stream: since streams are asynchronous, no events from the stream can be obtained before the initial build.

So changing the behavior of any `rxdart` stream will have no effect on this behavior.
